packages/react/src/dialog/Dialog.tsx

```diff
@@ -0,0 +1,60 @@
+import * as RadixDialog from "@radix-ui/react-dialog";
+import { type ComponentPropsWithRef, type ReactNode, forwardRef } from "react";
+
+import type { ExtendProps } from "../utils";
+
+import { Box } from "../box";
+import { Button } from "../button";
+import { Flex } from "../flex";
+import { Paper } from "../paper";
+import { CloseIcon } from "./CloseIcon";
+import * as styles from "./Dialog.css";
+
+type DialogProps = ExtendProps<
+  ComponentPropsWithRef<typeof Box>,
+  ComponentPropsWithRef<typeof RadixDialog.Root>,
+  {
+    children: ReactNode;
+    onClose: () => void;
+    withCloseButton?: boolean;
+  } & styles.DialogVariants
+>;
+export const Dialog = forwardRef<HTMLDivElement, DialogProps>(
+  (
+    { children, onClose, size = "md", withCloseButton = false, ...props },
+    ref,
+  ) => {
+    return (
+      <Box asChild ref={ref} {...props}>
+        <RadixDialog.Root onOpenChange={onClose}>
+          <Box asChild>
+            <RadixDialog.Portal>
+              <RadixDialog.Overlay asChild>
+                <Flex {...styles.overlay()}>
```

Comment:
sorry - actually can we go back to the previous usage with the flex outside? that one is more accurate since the radix component should have priority.

the difference is kind of like saying `interface Overlay extends Flex` rather than `interface Flex extends Overlay`

```suggestion
              <Flex asChild {...styles.overlay()}>
                <RadixDialog.Overlay>
```